libraries/src/Helper/ContentHistoryHelper.php

```
@@ -59,10 +60,16 @@ public function deleteHistory($table)
 		$typeTable = Table::getInstance('Contenttype', 'JTable');
 		$typeId = $typeTable->getTypeId($this->typeAlias);
 		$db = Factory::getDbo();
-		$query = $db->getQuery(true);
-		$query->delete($db->quoteName('#__ucm_history'))
-			->where($db->quoteName('ucm_item_id') . ' = ' . (int) $id)
-			->where($db->quoteName('ucm_type_id') . ' = ' . (int) $typeId);
+		$query = $db->getQuery(true)
+			->delete($db->quoteName('#__ucm_history'))
+			->where(
+				[
+					$db->quoteName('ucm_item_id') . ' = :id',
+					$db->quoteName('ucm_type_id') . ' = :typeId',
+				]
+			)
+			->bind(':id', $id, ParameterType::INTEGER)
```

Don't we need to typecast to (int)  `$typeId` and `$id `  like as is done in line 91-92 ?

---

JTable should return an int because the primary key is an integer, beside this is that the this query (maybe the complete feature) doesn't support non integer keys or multi column keys. I think type cast is not needed.

CS should be fixed.

---

If someone does try using a table with string PK here, it would error on PgSQL. I guess we could cast to int here to have it fail silently like on MySQL.

---

In this case it would try to delete entry 0 (zero) which is not better. But maybe out of scope of this PR. please cast it to int so we don't have a behavior change. There is another PR dealing with content versioning and ucm.

---

OK.